sql/expression_value.cc

```diff
@@ -3292,15 +3315,42 @@ void
 ExpressionValue::
 mergeToRowDestructive(StructValue & row)
 {
-    row.reserve(row.size() + rowLength());
+    // This is optimized here because the SelectExpression uses it whenever
+    // merging
+    switch (type_) {
+    case Type::STRUCTURED:
+        if (structured_.unique()) {
+            // We have the only reference in the shared pointer, AND we have
```

Comment:
Yes... That's the whole point of unique ().  Note that this is a non const method and so only this thread can be in a method now.  It's used in several other places.